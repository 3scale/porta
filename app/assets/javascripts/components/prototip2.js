

var Bubble = Class.create();
Bubble.prototype = {
  initialize: function(el) {
    this.offsetX = -8;
    this.offsetY = 38;

    this.target = el;
    // this.text = text;
    this.makeBubble();
  },
 
  makeBubble: function() {
    var bubble = $("bubble_" + this.target.title);
    // var bubble = new Element('div', { 'class': 'bubble'});
    // var content = new Element('div', { 'class': 'content'});
    // content.innerHTML = this.text;

    // bubble.appendChild(content);
    this.bubble = bubble;
    // new Insertion.Bottom(body, bubble);
    // bubble.hide();
    
    
    
    // set position

    // listen!
    
    this.target.observe('mouseover', (
        function() { 

          var position = this.target.cumulativeOffset();

          this.bubble.style.top = position[1] + this.offsetX + "px";
          this.bubble.style.left = position[0] + this.target.getWidth() + this.offsetY + "px";
          
          this.bubble.show(); 
        }).bind(this));
        
    this.target.observe('mouseout', (
        function() { 
          this.bubble.hide(); 
        }).bind(this));
        
  }
  
  
}

var BubbleHelper = {
  set: function(){
    $$('.define').each(function(l){
      new Bubble(l);      
    });
}}
  //     function(event) { 
  // 
  //       if ((event.element().className == 'submit' )) {
  // 
  //         var form = event.element().up('form');
  //         var formData = form.serialize(true);
  //         var url = form.action;
  //         
  //         this.ajaxPost(formData, url);
  //         event.stop();
  //       }
  //     })
  //   )        
  //   
  // }
