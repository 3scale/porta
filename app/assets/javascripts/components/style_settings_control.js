// Functions dealing with provider side control of site colours. Probably to be deprecated with the
// development of CMS system

jQuery.noConflict();

var settings = function() {
    // var B = ["background_color", "text_color", "link_color", "sidebar_fill_color", "sidebar_border_color"];
    var A = {
        updateColors: function() {
            var E = jQuery("#settings_bg_colour").val();
            var G = jQuery("#settings_text_colour").val();
            var F = jQuery("#settings_link_colour").val();
            var C = jQuery("#settings_menu_bg_colour").val();
						var D = jQuery("#settings_menu_link_colour").val();
						var H = jQuery("#settings_content_bg_colour").val();
						var J = jQuery("#settings_plans_tab_bg_colour").val();
						var K = jQuery("#settings_content_border_colour").val();
						
            jQuery("#previewContent").css({"background-color": E, color: G});

						jQuery('#pBox span').css("color", G);
						jQuery('#pBox  span a').css("color", F);
						jQuery('#pLinks span.active').css("background-color", H);
						jQuery('#pLinks span').css("color", D);
						jQuery('#pBox, #pFooter').css("background-color", H);
						jQuery('#pBox, #pFooter').css("border-color", K);
						jQuery('#previewPlans table th.head').css("background-color", J);
						
            jQuery("#pMenu").css({
                "background-color": C
            });
        },
        synchColors: function(D) {
            var C = settings.colorPicker.linked;
            if (C && C.value && C.value != settings.colorPicker.color) {
                C.value = settings.colorPicker.color;
                jQuery(C).css({
                    "background-color": C.value,
                    color: settings.colorPicker.hsl[2] > 0.5 ? "#000": "#fff"
                });
                settings.updateColors()
            }
        },
        initDesignSettings: function() {

          jQuery("#linkDesign a").click(function() {
              jQuery("#designSettings").show();
							jQuery('#detailSettings').hide();
              jQuery("#linkDetails a").removeClass("active");
              jQuery("#linkDesign a").addClass("active");
              jQuery("#linkDetails").removeClass("active");
              jQuery("#linkDesign").addClass("active");
							
              return false
          });

          jQuery("#linkDetails a").click(function() {
              jQuery("#designSettings").hide();
							jQuery('#detailSettings').show();

              // jQuery("#backgrounds").hide();
              jQuery("#linkDetails a").addClass("active");
              jQuery("#linkDesign a").removeClass("active");
              jQuery("#linkDetails").addClass("active");
              jQuery("#linkDesign").removeClass("active");

              // if (jQuery("#colors").css("display") == "none") {
              //     jQuery("#title_picker").removeClass("active");
              //     jQuery("#title_background").removeClass("inactive")
              // } else {
              //     jQuery("#title_picker").addClass("active");
              //     jQuery("#title_background").addClass("inactive")
              // }
              // if (jQuery("#user_profile_default").val()) {
              //     jQuery("#user_profile_default").val(false)
              // }
              return false
          });

            jQuery("#title_picker a").click(function() {
                jQuery("#colors").toggle();
                jQuery("#backgrounds").hide();
                jQuery("#title_background").removeClass("active");
                if (jQuery("#colors").css("display") == "none") {
                    jQuery("#title_picker").removeClass("active");
                    jQuery("#title_background").removeClass("inactive")
                } else {
                    jQuery("#title_picker").addClass("active");
                    jQuery("#title_background").addClass("inactive")
                }
                if (jQuery("#user_profile_default").val()) {
                    jQuery("#user_profile_default").val(false)
                }
                return false
            });
            jQuery("#title_background a").click(function() {
                jQuery("#backgrounds").toggle();
                jQuery("#colors").hide();
                jQuery("#title_picker").removeClass("active");
                if (jQuery("#backgrounds").css("display") == "none") {
                    jQuery("#title_background").removeClass("active");
                    jQuery("#title_picker").removeClass("inactive")
                } else {
                    jQuery("#title_background").addClass("active");
                    jQuery("#title_picker").addClass("inactive")
                }
                return false
            });
            jQuery("#themes a").click(function() {
                var E = jQuery(this).attr("id");
                settings.setTheme(E.substring(5));
                return false
            });
            jQuery("#backgrounds a").click(function() {
                jQuery("#backgrounds a").removeClass("active");
                jQuery(this).addClass("active");
                var E = jQuery(this).attr("id");
                if (E == "current_background") {
                    settings.setBackgroundImage(_tmp.currentImage)
                } else {
                    if (E == "no_background") {
                        settings.setBackgroundImage("none")
                    }
                }
                return false
            });
            jQuery("#user_profile_background_tile").click(function() {
                var E = jQuery("#user_profile_background_tile:checked").val();
                jQuery("body").css({
                    "background-repeat": E ? "repeat": "no-repeat",
                    "background-attachment": E ? "scroll": "fixed"
                });
                jQuery("#user_profile_use_background_image").val("true")
            });
            settings.colorPicker = jQuery.farbtastic("#picker", settings.synchColors);
            jQuery("#picker").click(function() {
                jQuery(".colorwell-selected").focus()
            });
            var D;
            var C = (function() {
                var E = [8, 9, 13, 16, 17, 18, 37, 38, 39, 40, 45, 46, 86, 88, 90];
                for (var F = 48; F <= 57; F++) {
                    E.push(F)
                }
                for (var F = 65; F <= 70; F++) {
                    E.push(F)
                }
                for (var F = 96; F <= 105; F++) {
                    E.push(F)
                }
                return E
            })();
            jQuery(".colorwell").each(settings.synchColors).focus(function() {
                if (D) {
                    jQuery(D).removeClass("colorwell-selected")
                }
                settings.colorPicker.linked = this;
                settings.colorPicker.setColor(this.value);
                jQuery("#picker").show();
                jQuery(D = this).addClass("colorwell-selected")
            }).blur(function() {
                var E = jQuery(E);
                E.val(E.val().toUpperCase())
            }).keydown(function(E) {
                if (jQuery.inArray(E.keyCode, C) != -1) {
                    if (jQuery.inArray(E.keyCode, [86, 88, 90]) != -1) {
                        return (E.ctrlKey || E.metaKey)
                    } else {
                        if ((jQuery.inArray(E.keyCode, [8, 46]) != -1) || ((E.keyCode == 88) && E.shiftKey)) {
                            return (jQuery(this).val() != "#")
                        } else {
                            if (E.shiftKey) {
                                return (jQuery.inArray(E.keyCode, [65, 66, 67, 68, 69, 70]) != -1)
                            }
                        }
                    }
                    return true
                }
                return false
            }).keyup(function(E) {
                if (jQuery(this).val().indexOf("#") != 0) {
                    jQuery(this).val("#" + jQuery(this).val())
                }
                settings.colorPicker.setColor(this.value);
                var F = jQuery(this).val();
                if (!document.all || (F.length == 7 || F.length == 4)) {
                    jQuery(this).css({
                        "background-color": jQuery(this).val(),
                        color: settings.colorPicker.hsl[2] > 0.5 ? "#000000": "#ffffff"
                    });
                    settings.updateColors()
                }
            });
            jQuery("#design_customization form a").click(settings.updateColors)
        }
    };
    return A
} ();

jQuery(document).ready(function() {
		if($('settingsBox')) 
    	settings.initDesignSettings();
});
