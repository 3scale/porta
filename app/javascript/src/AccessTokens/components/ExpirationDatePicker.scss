@import '~@patternfly/patternfly/patternfly-addons.css';

.pf-c-calendar-month,
.pf-c-form-control-expiration {
  width: 50%;
}

.pf-c-form-control-expiration {
  margin-right: 1em;
}

button.pf-c-form__group-label-help {
  min-width: auto;
  width: auto;
}
