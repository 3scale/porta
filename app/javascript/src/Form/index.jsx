// @flow

export * from 'Form/FormFieldset'
export * from 'Form/FormLegend'
