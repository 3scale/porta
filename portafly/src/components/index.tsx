export * from './pages'
export * from './shared'
export * from './AppLayout'
export * from './AppNavExpandable'
export * from './AppNavGroup'
export * from './AppNavItem'
export * from './AppHeaderTools'
export * from './LazyRoute'
export * from './NotFound'
export * from './SwitchWith404'
export * from './util'
export * from './Root'
