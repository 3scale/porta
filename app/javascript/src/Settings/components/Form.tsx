import { useState } from 'react'

import {
  FormCollection,
  FormFieldset,
  FormLegend,
  TextInputGroup,
  RadioFieldset
} from 'Settings/components/Common'
import { AuthenticationSettingsFieldset } from 'Settings/components/AuthenticationSettingsFieldset'
import { createReactWrapper } from 'utilities/createReactWrapper'
import { SETTINGS_DEFAULT } from 'Settings/defaults'
import { FieldCatalogProps, FieldGroupProps, LegendCollectionProps, TypeItemProps } from 'Settings/types'

const SERVICE_MESH_INTEGRATION = 'service_mesh_istio'
const PROXY_HOSTED_INTEGRATION = 'hosted'

type Props = {
  isProxyCustomUrlActive: boolean,
  integrationMethod: FieldGroupProps & FieldCatalogProps,
  authenticationMethod: FieldGroupProps & FieldCatalogProps,
  proxyEndpoints: FieldGroupProps[],
  authenticationSettings: {
    appIdKeyPairSettings: FieldGroupProps[],
    apiKeySettings: FieldGroupProps,
    oidcSettings: {
      basicSettings: TypeItemProps,
      flowSettings: FieldGroupProps[],
      jwtSettings: TypeItemProps
    }
  },
  credentialsLocation: FieldGroupProps & FieldCatalogProps,
  security: FieldGroupProps[],
  gatewayResponse: LegendCollectionProps[]
}

const Form: React.FunctionComponent<Props> = ({
  isProxyCustomUrlActive,
  integrationMethod,
  authenticationMethod,
  proxyEndpoints,
  authenticationSettings,
  credentialsLocation,
  security,
  gatewayResponse
}) => {
  const [ selectedIntegrationMethod, setSelectedIntegrationMethod ] = useState(integrationMethod.value)
  const [ selectedAuthenticationMethod, setSelectedAuthenticationMethod ] = useState(authenticationMethod.value)
  const onChange = (setState: (arg1: ((arg1: string) => string) | string) => void) => (_checked: string, e: React.SyntheticEvent<HTMLButtonElement>) => setState(e.currentTarget.value)
  const isServiceMesh = selectedIntegrationMethod === SERVICE_MESH_INTEGRATION
  const isProxyHosted = selectedIntegrationMethod === PROXY_HOSTED_INTEGRATION
  const isProxyUrlsReadOnly = !isProxyCustomUrlActive && isProxyHosted
  const customProxyEndpoints = proxyEndpoints.map(endpoint => ({
    ...endpoint,
    readOnly: isProxyUrlsReadOnly,
    isDefaultValue: isProxyUrlsReadOnly
  }))

  return (
     <>
      <RadioFieldset {...integrationMethod} onChange={onChange(setSelectedIntegrationMethod)} value={selectedIntegrationMethod} legend='Integration' />
      { !isServiceMesh && <FormCollection collection={customProxyEndpoints} ItemComponent={TextInputGroup} legend='API gateway' /> }
      <RadioFieldset {...authenticationMethod} onChange={onChange(setSelectedAuthenticationMethod)} value={selectedAuthenticationMethod} legend='Authentication' />
      {/* $FlowIgnore[cannot-spread-inexact] */}
      <AuthenticationSettingsFieldset
        isServiceMesh={isServiceMesh}
        authenticationMethod={selectedAuthenticationMethod}
        {...authenticationSettings}
      />
      { !isServiceMesh &&  <>
        <RadioFieldset {...credentialsLocation} legend='Credentials Location' />
        <FormCollection collection={security} ItemComponent={TextInputGroup} legend='Security' />
        <FormFieldset id='fieldset-GatewayResponse'>
          <FormLegend>Gateway Response</FormLegend>
          {gatewayResponse.map(settings => (
            <FormCollection key={settings.legend} collection={settings.collection} ItemComponent={TextInputGroup} legend={settings.legend} />
          ))}
        </FormFieldset>
       </> }
     </>
  )
}

const FormWrapper = (settings: Props = SETTINGS_DEFAULT, elementId: string) => createReactWrapper(<Form {...settings} />, elementId)

export { Form, FormWrapper, Props }
