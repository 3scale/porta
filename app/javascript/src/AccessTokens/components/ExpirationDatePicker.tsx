import { useState, useMemo } from 'react'
import { Alert, CalendarMonth, FormGroup, FormSelect, FormSelectOption } from '@patternfly/react-core'

import { createReactWrapper } from 'utilities/createReactWrapper'

import type { FunctionComponent, FormEvent } from 'react'

import './ExpirationDatePicker.scss'

interface ExpirationItem {
  id: string;
  label: string;
  period: number; // In seconds
}

const collection: ExpirationItem[] = [
  { id: '7', label: '7 days', period: 7 },
  { id: '30', label: '30 days', period: 30 },
  { id: '60', label: '60 days', period: 60 },
  { id: '90', label: '90 days', period: 90 },
  { id: 'custom', label: 'Custom...', period: 0 },
  { id: 'no-exp', label: 'No expiration', period: 0 }
]

const dayMs = 60 * 60 * 24 * 1000

interface Props {
  id: string;
  label: string | null;
}

const ExpirationDatePicker: FunctionComponent<Props> = ({ id, label }) => {
  const [selectedItem, setSelectedItem] = useState(collection[0])
  const [pickedDate, setPickedDate] = useState(new Date())
  const fieldName = `human_${id}`
  const fieldLabel = label ?? 'Expires in'

  const fieldDate = useMemo(() => {
    if (selectedItem.period === 0) return null

    return new Date(new Date().getTime() + selectedItem.period * dayMs)
  }, [selectedItem])

  const fieldHint = useMemo(() => {
    if (!fieldDate) return

    const date = new Date(fieldDate)
    date.setHours(0, 0, 0, 0)

    return `The token will expire on ${date.toLocaleDateString()}`
  }, [fieldDate])

  const dateValue = useMemo(() => {
    let value = ''

    if (fieldDate) {
      value = fieldDate.toISOString()
    } else if (selectedItem.id === 'custom' ) {
      value = pickedDate.toISOString()
    }

    return value
  }, [fieldDate, selectedItem, pickedDate])

  const handleOnChange = (_value: string, event: FormEvent<HTMLSelectElement>) => {
    const value = (event.target as HTMLSelectElement).value
    const selected = collection.find(i => i.id === value) ?? null

    if (selected === null) return

    setSelectedItem(selected)
    setPickedDate(new Date())
  }

  return (
    <>
      <FormGroup
        isRequired
        fieldId={fieldName}
        label={fieldLabel}
      >
        <FormSelect
          className="pf-c-form-control-expiration"
          id={fieldName}
          value={selectedItem.id}
          onChange={handleOnChange}
        >
          {collection.map((item: ExpirationItem) => {
            return (
              <FormSelectOption
                key={item.id}
                label={item.label}
                value={item.id}
              />
            )
          })}
        </FormSelect>
        <span className="pf-c-form-control-expiration-hint">{fieldHint}</span>
      </FormGroup>
      <input id={id} name={id} type="hidden" value={dateValue} />
      {selectedItem.id === 'custom' && (
        <CalendarMonth className="pf-u-mt-md" date={pickedDate} onChange={setPickedDate} />
      )}
      {dateValue === '' && (
        <>
          <br />
          <Alert title="Expiration is recommended" variant="warning">
            It is strongly recommended that you set an expiration date for your token to help keep your information
            secure
          </Alert>
        </>
      )}
    </>
  )
}

const ExpirationDatePickerWrapper = (props: Props, containerId: string): void => { createReactWrapper(<ExpirationDatePicker id={props.id} label={props.label} />, containerId) }

export type { ExpirationItem, Props }
export { ExpirationDatePicker, ExpirationDatePickerWrapper }
