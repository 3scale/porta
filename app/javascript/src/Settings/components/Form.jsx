// @flow

import * as React from 'react'
import { useState } from 'react'
import { FormFieldset, FormLegend } from 'Form'
import { FormCollection, TextInputGroup, RadioFieldset } from 'Settings/components/Common'
import { AuthenticationSettingsFieldset } from 'Settings/components/AuthenticationSettingsFieldset'
import type { SettingsProps as Props } from 'Settings/types'

const SERVICE_MESH_INTEGRATION = 'service_mesh_istio'
const PROXY_HOSTED_INTEGRATION = 'hosted'

const Form = ({
  isProxyCustomUrlActive,
  integrationMethod,
  authenticationMethod,
  proxyEndpoints,
  authenticationSettings,
  credentialsLocation,
  security,
  gatewayResponse
}: Props): React.Node => {
  const [ selectedIntegrationMethod, setSelectedIntegrationMethod ] = useState(integrationMethod.value)
  const [ selectedAuthenticationMethod, setSelectedAuthenticationMethod ] = useState(authenticationMethod.value)
  const onChange = (setState) => (_checked, e) => setState(e.currentTarget.value)
  const isServiceMesh = selectedIntegrationMethod === SERVICE_MESH_INTEGRATION
  const isProxyHosted = selectedIntegrationMethod === PROXY_HOSTED_INTEGRATION
  const isProxyUrlsReadOnly = !isProxyCustomUrlActive && isProxyHosted
  const customProxyEndpoints = proxyEndpoints.map(endpoint =>
    ({ ...endpoint, readOnly: isProxyUrlsReadOnly, isDefaultValue: isProxyUrlsReadOnly }))

  return (
    <React.Fragment>
      <RadioFieldset {...integrationMethod} onChange={onChange(setSelectedIntegrationMethod)} value={selectedIntegrationMethod} legend='Integration' />
      { !isServiceMesh && <FormCollection collection={customProxyEndpoints} ItemComponent={TextInputGroup} legend='API gateway' /> }
      <RadioFieldset {...authenticationMethod} onChange={onChange(setSelectedAuthenticationMethod)} value={selectedAuthenticationMethod} legend='Authentication' />
      {/* $FlowIgnore[cannot-spread-inexact] */}
      <AuthenticationSettingsFieldset
        isServiceMesh={isServiceMesh}
        authenticationMethod={selectedAuthenticationMethod}
        {...authenticationSettings}
      />
      { !isServiceMesh && <React.Fragment>
        <RadioFieldset {...credentialsLocation} legend='Credentials Location' />
        <FormCollection collection={security} ItemComponent={TextInputGroup} legend='Security' />
        <FormFieldset id='fieldset-GatewayResponse'>
          <FormLegend>Gateway Response</FormLegend>
          {gatewayResponse.map(settings => (
            <FormCollection key={settings.legend} collection={settings.collection} ItemComponent={TextInputGroup} legend={settings.legend} />
          ))}
        </FormFieldset>
      </React.Fragment> }
    </React.Fragment>
  )
}

export {
  Form
}
