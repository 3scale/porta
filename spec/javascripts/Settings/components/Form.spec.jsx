import React from 'react'
import { shallow } from 'enzyme'
import { Form } from 'Settings/components/Form'
import {
  SETTINGS_DEFAULT,
  INTEGRATION_METHOD_DEFAULTS
} from 'Settings/defaults'

function setup (customProps = {}) {
  const props = {
    ...SETTINGS_DEFAULT,
    ...customProps
  }

  const view = shallow(<Form {...props} />)

  return { view, props }
}

it('should render correctly', () => {
  const { view } = setup()
  expect(view).toMatchSnapshot()
})

it('should not render Auth Settings, Security, Credential Locations and API Gateway when Istio is selected', () => {
  const customProps = { integrationMethod: {...INTEGRATION_METHOD_DEFAULTS, value: 'service_mesh_istio'} }
  const { view } = setup(customProps)
  expect(view).toMatchSnapshot()
})
