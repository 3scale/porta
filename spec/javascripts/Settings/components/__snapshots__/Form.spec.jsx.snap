// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render Auth Settings, Security, Credential Locations and API Gateway when Istio is selected 1`] = `
<Fragment>
  <RadioFieldset
    catalog={
      Object {
        "hosted": "APIcast 3scale managed",
        "self_managed": "APIcast self-managed",
        "service_mesh_istio": "Istio",
      }
    }
    legend="Integration"
    name="deployment_option"
    onChange={[Function]}
    value="service_mesh_istio"
  />
  <RadioFieldset
    catalog={
      Object {
        "1": "API Key (user_key)",
        "2": "App_ID and App_Key Pair",
        "oidc": "OpenID Connect",
      }
    }
    legend="Authentication"
    name="proxy_authentication_method"
    onChange={[Function]}
    value="1"
  />
  <AuthenticationSettingsFieldset
    apiKeySettings={
      Object {
        "label": "Auth user key",
        "name": "auth_user_key",
        "value": "user_key",
      }
    }
    appIdKeyPairSettings={
      Array [
        Object {
          "hint": "Public address of your API gateway in the staging environment.",
          "label": "App ID parameter",
          "name": "auth_app_id",
          "value": "app_id",
        },
        Object {
          "label": "App Key parameter",
          "name": "auth_app_key",
          "value": "app_key",
        },
      ]
    }
    authenticationMethod="1"
    isServiceMesh={true}
    oidcSettings={
      Object {
        "basicSettings": Object {
          "item": Object {
            "hint": "Location of your OpenID Provider. The format of this endpoint is determined on your OpenID Provider setup. A common guidance would be \\"https://CLIENT_ID:CLIENT_SECRET@HOST:PORT/auth/realms/REALM_NAME\\".",
            "label": "OpenID Connect Issuer",
            "name": "oidc_issuer_endpoint",
            "placeholder": "https://sso.example.com/auth/realms/gateway",
            "value": "",
          },
          "type": Object {
            "catalog": Object {
              "keycloak": "Red Hat Single Sign-On",
              "rest": "REST API",
            },
            "label": "OpenID Connect Issuer Type",
            "name": "oidc_issuer_type",
            "value": "keycloak",
          },
        },
        "flowSettings": Array [
          Object {
            "checked": false,
            "label": "Service Accounts Flow",
            "name": "service_accounts_enabled",
          },
          Object {
            "checked": false,
            "label": "Authorization Code Flow",
            "name": "standard_flow_enabled",
          },
          Object {
            "checked": false,
            "label": "Implicit Flow",
            "name": "implicit_flow_enabled",
          },
          Object {
            "checked": false,
            "label": "Direct Access Grant Flow",
            "name": "direct_access_grants_enabled",
          },
        ],
        "jwtSettings": Object {
          "item": Object {
            "hint": "The Token Claim that contains the clientID. Defaults to \\"azp\\".",
            "label": "ClientID Token Claim",
            "name": "jwt_claim_with_client_id",
            "placeholder": "azp",
            "value": "azp",
          },
          "type": Object {
            "catalog": Object {
              "liquid": "liquid",
              "plain": "plain",
            },
            "hint": "Process the ClientID Token Claim value as a string or as a liquid template. When set to \\"Liquid\\" you can define more complex rules. e.g. If \\"some_claim\\" is an array you can select the first value this like {{ some_claim | first }}.",
            "label": "ClientID Token Claim Type",
            "name": "jwt_claim_with_client_id_type",
            "value": "plain",
          },
        },
      }
    }
  />
</Fragment>
`;

exports[`should render correctly 1`] = `
<Fragment>
  <RadioFieldset
    catalog={
      Object {
        "hosted": "APIcast 3scale managed",
        "self_managed": "APIcast self-managed",
        "service_mesh_istio": "Istio",
      }
    }
    legend="Integration"
    name="deployment_option"
    onChange={[Function]}
    value="hosted"
  />
  <FormCollection
    ItemComponent={[Function]}
    collection={
      Array [
        Object {
          "defaultValue": "https://api-2.staging.apicast.com",
          "hint": "Public address of your API gateway in the staging environment.",
          "inputType": "url",
          "isDefaultValue": true,
          "label": "Staging Public Base URL",
          "name": "sandbox_endpoint",
          "placeholder": "https://api.provider-name.com",
          "readOnly": true,
          "value": "https://custom.api.staging.provider-name.com",
        },
        Object {
          "defaultValue": "https://api-2.apicast.com",
          "hint": "Public address of your API gateway in the production environment.",
          "inputType": "url",
          "isDefaultValue": true,
          "label": "Staging Public Base URL",
          "name": "endpoint",
          "placeholder": "https://api.provider-name.com",
          "readOnly": true,
          "value": "https://custom.api.provider-name.com",
        },
      ]
    }
    legend="API gateway"
  />
  <RadioFieldset
    catalog={
      Object {
        "1": "API Key (user_key)",
        "2": "App_ID and App_Key Pair",
        "oidc": "OpenID Connect",
      }
    }
    legend="Authentication"
    name="proxy_authentication_method"
    onChange={[Function]}
    value="1"
  />
  <AuthenticationSettingsFieldset
    apiKeySettings={
      Object {
        "label": "Auth user key",
        "name": "auth_user_key",
        "value": "user_key",
      }
    }
    appIdKeyPairSettings={
      Array [
        Object {
          "hint": "Public address of your API gateway in the staging environment.",
          "label": "App ID parameter",
          "name": "auth_app_id",
          "value": "app_id",
        },
        Object {
          "label": "App Key parameter",
          "name": "auth_app_key",
          "value": "app_key",
        },
      ]
    }
    authenticationMethod="1"
    isServiceMesh={false}
    oidcSettings={
      Object {
        "basicSettings": Object {
          "item": Object {
            "hint": "Location of your OpenID Provider. The format of this endpoint is determined on your OpenID Provider setup. A common guidance would be \\"https://CLIENT_ID:CLIENT_SECRET@HOST:PORT/auth/realms/REALM_NAME\\".",
            "label": "OpenID Connect Issuer",
            "name": "oidc_issuer_endpoint",
            "placeholder": "https://sso.example.com/auth/realms/gateway",
            "value": "",
          },
          "type": Object {
            "catalog": Object {
              "keycloak": "Red Hat Single Sign-On",
              "rest": "REST API",
            },
            "label": "OpenID Connect Issuer Type",
            "name": "oidc_issuer_type",
            "value": "keycloak",
          },
        },
        "flowSettings": Array [
          Object {
            "checked": false,
            "label": "Service Accounts Flow",
            "name": "service_accounts_enabled",
          },
          Object {
            "checked": false,
            "label": "Authorization Code Flow",
            "name": "standard_flow_enabled",
          },
          Object {
            "checked": false,
            "label": "Implicit Flow",
            "name": "implicit_flow_enabled",
          },
          Object {
            "checked": false,
            "label": "Direct Access Grant Flow",
            "name": "direct_access_grants_enabled",
          },
        ],
        "jwtSettings": Object {
          "item": Object {
            "hint": "The Token Claim that contains the clientID. Defaults to \\"azp\\".",
            "label": "ClientID Token Claim",
            "name": "jwt_claim_with_client_id",
            "placeholder": "azp",
            "value": "azp",
          },
          "type": Object {
            "catalog": Object {
              "liquid": "liquid",
              "plain": "plain",
            },
            "hint": "Process the ClientID Token Claim value as a string or as a liquid template. When set to \\"Liquid\\" you can define more complex rules. e.g. If \\"some_claim\\" is an array you can select the first value this like {{ some_claim | first }}.",
            "label": "ClientID Token Claim Type",
            "name": "jwt_claim_with_client_id_type",
            "value": "plain",
          },
        },
      }
    }
  />
  <RadioFieldset
    catalog={
      Object {
        "authorization": "As HTTP Basic Authentication",
        "headers": "As HTTP Headers",
        "query": "As query parameters (GET) or body parameters (POST/PUT/DELETE)",
      }
    }
    legend="Credentials Location"
    name="credentials_location"
    value="headers"
  />
  <FormCollection
    ItemComponent={[Function]}
    collection={
      Array [
        Object {
          "defaultValue": "",
          "hint": "Lets you define a custom Host request header. This is needed if your API backend only accepts traffic from a specific host.",
          "label": "Host Header",
          "name": "hostname_rewrite",
          "placeholder": "https://api.provider-name.com",
          "readOnly": false,
          "value": "",
        },
        Object {
          "defaultValue": "",
          "hint": "Enables you to block any direct developer requests to your API backend; each 3scale API gateway call to your API backend contains a request header called X-3scale-proxy-secret-token. The value of this header can be set by you here. It's up to you ensure your backend only allows calls with this secret header.",
          "label": "Secret Token",
          "name": "secret_token",
          "placeholder": "https://api.provider-name.com",
          "readOnly": false,
          "value": "",
        },
      ]
    }
    legend="Security"
  />
  <FormFieldset
    id="fieldset-GatewayResponse"
  >
    <FormLegend>
      Gateway Response
    </FormLegend>
    <FormCollection
      ItemComponent={[Function]}
      collection={
        Array [
          Object {
            "inputType": "number",
            "label": "Response Code",
            "name": "error_status_auth_failed",
            "value": "403",
          },
          Object {
            "inputType": "text",
            "label": "Content-type",
            "name": "error_headers_auth_failed",
            "value": "text/plain; charset=us-ascii",
          },
          Object {
            "inputType": "text",
            "label": "Response Body",
            "name": "error_auth_failed",
            "value": "Authentication failed",
          },
        ]
      }
      key="Authentication Failed Error"
      legend="Authentication Failed Error"
    />
    <FormCollection
      ItemComponent={[Function]}
      collection={
        Array [
          Object {
            "inputType": "number",
            "label": "Response Code",
            "name": "error_status_auth_missing",
            "value": "403",
          },
          Object {
            "inputType": "text",
            "label": "Content-type",
            "name": "error_headers_auth_missing",
            "value": "text/plain; charset=us-ascii",
          },
          Object {
            "inputType": "text",
            "label": "Response Body",
            "name": "error_auth_missing",
            "value": "Authentication parameters missing",
          },
        ]
      }
      key="Authentication Missing Error"
      legend="Authentication Missing Error"
    />
    <FormCollection
      ItemComponent={[Function]}
      collection={
        Array [
          Object {
            "inputType": "number",
            "label": "Response Code",
            "name": "error_status_no_match",
            "value": "404",
          },
          Object {
            "inputType": "text",
            "label": "Content-type",
            "name": "error_headers_no_match",
            "value": "text/plain; charset=us-ascii",
          },
          Object {
            "inputType": "text",
            "label": "Response Body",
            "name": "error_no_match",
            "value": "No Mapping Rule matched",
          },
        ]
      }
      key="Match Error"
      legend="Match Error"
    />
    <FormCollection
      ItemComponent={[Function]}
      collection={
        Array [
          Object {
            "inputType": "number",
            "label": "Response Code",
            "name": "error_status_limits_exceeded",
            "value": "429",
          },
          Object {
            "inputType": "text",
            "label": "Content-type",
            "name": "error_headers_limits_exceeded",
            "value": "text/plain; charset=us-ascii",
          },
          Object {
            "inputType": "text",
            "label": "Response Body",
            "name": "error_limits_exceeded",
            "value": "Usage limit exceeded",
          },
        ]
      }
      key="Usage limit exceeded error"
      legend="Usage limit exceeded error"
    />
  </FormFieldset>
</Fragment>
`;
